### PR TITLE
importer schema conversion fix: don't use overly coarse resolution data just because it has a long retention

### PR DESCRIPTION
**Pull request by replay**
fixes: #1596

### CODE REVIEW

File: mdata/importer/converter.go [hunk 1]
```diff
@@ -132,6 +122,15 @@ func (c *converter) findSmallestLargestArchive(spp, nop uint32) (int, int) {
 		smallestArchiveIdx = i
 	}
 
+	// find smallest archive that still contains enough data to satisfy requested range
```
Comment (Dieterbe) [inline mdata/importer/converter.go]: if smallest already points to an archive that has enough TTL, then that's the only thing we need. we don't need any other archives

Comment (replay) [reply mdata/importer/converter.go]: yep, that's why now it's starting to check from smallest and if it has enough TTL then it uses smallest as largest as well

**Comment by fitzoh** [comment]
Would something like this make sense?

```go
func (c *converter) findSmallestLargestArchive(spp, nop uint32) (int, int) {
	smallestArchiveIdx := -1
	largestArchiveIdx := -1
	threshold := nop * spp
	for i, arch := range c.archives {
		// find largest archive that still has a higher or equal resolution than requested
		if arch.SecondsPerPoint <= spp {
			smallestArchiveIdx = i
		}
		// find smallest archive that still contains enough data to satisfy requested range,
		if arch.SecondsPerPoint > threshold {
			break
		}
		largestArchiveIdx = i
	}

	return smallestArchiveIdx, largestArchiveIdx
}
```

**Comment by replay** [comment]
thx @fitzoh that looks good. 
at some point we should also fix this schema conversion so that it solves this issue as well, but i guess that could be done in a new PR: https://github.com/grafana/metrictank/issues/1609

**Comment by Dieterbe** [comment]
I rewrote the history because it was a mess, and came up with a clearer title.

**Pull request merged**